### ISSUE TITLE
TaggableManager blank=False by default.

### ISSUE THREAD

**Issue by soulne4ny** [jazzband/django-taggit#127]
It is really strange that default `blank` is `False` with respect that default `null` is `True`: https://github.com/alex/django-taggit/blob/master/taggit/managers.py#L52

It does not seem useful to have the `blank` option for the `TaggableManager`, does it?  What could be a reason to require at least one tag on a record?

**Comment by rtpg** [comment]
I guess the deeper issue is that `blank` is settable but `null` is not. I guess that means that you at least need to pass in an empty list for certain ops, but.... this one needs to be dug into a bit more for sure

**Comment by mattions** [comment]
I'm trying to use django-taggit with the django rest framework.
I have set the `blank=True` in the `TaggableManager`

```
# models.py 
Class MyModel(modes.Model):
    tags = TaggableManager(blank=True)
    
# serializers.py
class MySerializer(TaggitSerializer, serializers.HyperlinkedModelSerializer):

    tags = TagListSerializerField()
    class Meta:
        model = MyModel
```
but when I try to do a post, I receive the error:

```
{'tags': ['This field may not be blank.']}
```

Not sure if this is the right issue, or should I open a new one. 
Thanks for the advice!

**Comment by rtpg** [comment]
@mattions yeah so off the top of my head I think the solution for you is to put `blank=True` into `TagListserializerField()`. Basically the `TagListSerializerField` being declared explicitly means DRF uses what you provided in the initialization, no matter what you have in the underlying model.

Instead, what you can do (bit roundabout):

- [customize the field mappings in your serializers](https://www.django-rest-framework.org/api-guide/serializers/#customizing-field-mappings)  (you probably want to make a base serializer class for your whole project, generally a good idea anyways). You can make a mapping from `TaggableManager` to `TagListSerializerField`.
- In `MySerializer`, instead of declaring `tags` how you did, you can do 

```
class Meta:
  model = MyModel
  fields = ['tags']
```

This might or might not work. 

This is a pretty good FAQ item to be added honestly.

**Comment by Ryan0108** [comment]
Hi everybody, 

I'm stuck with the same problem...!

I found that a method is filtering the empty array before sending it to the manager, but i cannot find which one, any idea ? 

Thanks !

**Comment by Ryan0108** [comment]
Seems that the bug is in the serializer: 

```
def to_internal_value(self, value):
        # note to future maintainers: this field used to not be a ListField
        # and has extra behavior to support string-based input.
        #
        # In the future we should look at removing this feature so we can
        # make this a simple ListField (if feasible)
        print(f"Received value: {value}")
        if isinstance(value, list):
            if len(value) == 0:
                return ['test']
       .....
```

->Replace tags by test. However, an empty list is not sent to the "set" method in the manager ! 

Progressing...

**Comment by Ryan0108** [comment]
Seems that the key is in the update taggutSerializer:

    

> def update(self, instance, validated_data):
>         user = self.context["request"].user
>         tag_names = validated_data.pop("tags") if "tags" in validated_data else None
>         instance = super(TaggitSerializer, self).update(instance, validated_data)
>         if tag_names is not None:
>             tags = []
>             for name in tag_names:
>                 tag, created = FullTag.objects.get_or_create(
>                     name=name, specialite=user.profile.specialite_active
>                 )
>                 if created:
>                     # add user
>                     tag.user.add(user)
>                     # add mtom specialite
>                     tag.specialites.add(user.profile.specialite_active)
>                 else:
>                     # Check if user is already associated with the tag
>                     if not tag.user.filter(id=user.id).exists():
>                         tag.user.add(user)
>                     # Check if specialite is already associated with the tag
>                     if not tag.specialites.filter(
>                         id=user.profile.specialite_active.id
>                     ).exists():
>                         tag.specialites.add(user.profile.specialite_active)
> 
>                 tags.append(tag)
> 
>             instance.tags.set(tags, clear=True)
>         return instance
